test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -0,0 +1,386 @@
+package games.strategy.engine.data;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+
+import org.junit.Test;
+
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.util.IntegerMap;
+
+public class UnitCollectionTest {
+
+  final GameData gameData = new MyGameData();
+  final UnitType unitTypeOne = new UnitType("Unit Type 1", gameData);
+  final UnitType unitTypeTwo = new UnitType("Unit Type 2", gameData);
+  final MyPlayerID defaultPlayerID = new MyPlayerID();
+  final PlayerID otherPlayerID = PlayerID.NULL_PLAYERID;
+  final UnitCollection uc = new UnitCollection(defaultPlayerID, gameData);
+
+  final int unitCountDefaultPlayerUnitTypeOne = getUnitsOfDefaultPlayerOfUnitTypeOne().size();
+  final int unitCountDefaultPlayerUnitTypeTwo = getUnitsOfDefaultPlayerOfUnitTypeTwo().size();
+  final int unitCountDefaultPlayer = unitCountDefaultPlayerUnitTypeOne + unitCountDefaultPlayerUnitTypeTwo;
+  final Unit unitDefaultPlayer1 = new Unit(unitTypeOne, defaultPlayerID, gameData);
+  final Unit unitDefaultPlayer2 = new Unit(unitTypeTwo, defaultPlayerID, gameData);
+  final Unit unitDefaultPlayer3 = new Unit(unitTypeTwo, defaultPlayerID, gameData);
+
+  private Collection<Unit> getUnitsOfDefaultPlayerOfUnitTypeOne() {
+    final Collection<Unit> units = new ArrayList<Unit>();
+    units.add(unitDefaultPlayer1);
+    return units;
+  }
+
+  private Collection<Unit> getUnitsOfDefaultPlayerOfUnitTypeTwo() {
+    final Collection<Unit> units = new ArrayList<Unit>();
+    units.add(unitDefaultPlayer2);
+    units.add(unitDefaultPlayer3);
+    return units;
+  }
+
+  final int unitCountOtherPlayerUnitTypeOne = 3;
+  final int unitCountOtherPlayerUnitTypeTwo = 4;
+  final int unitCountOtherPlayer = unitCountOtherPlayerUnitTypeOne + unitCountOtherPlayerUnitTypeTwo;
+  final Unit unitOtherPlayer1 = new Unit(unitTypeOne, otherPlayerID, gameData);
+  final Unit unitOtherPlayer2 = new Unit(unitTypeOne, otherPlayerID, gameData);
+  final Unit unitOtherPlayer3 = new Unit(unitTypeOne, otherPlayerID, gameData);
+  final Unit unitOtherPlayer4 = new Unit(unitTypeTwo, otherPlayerID, gameData);
+  final Unit unitOtherPlayer5 = new Unit(unitTypeTwo, otherPlayerID, gameData);
+  final Unit unitOtherPlayer6 = new Unit(unitTypeTwo, otherPlayerID, gameData);
+  final Unit unitOtherPlayer7 = new Unit(unitTypeTwo, otherPlayerID, gameData);
+
+
+  final int unitCountUnitTypeOne = unitCountDefaultPlayerUnitTypeOne + unitCountOtherPlayerUnitTypeOne;
+  final int unitCountUnitTypeTwo = unitCountDefaultPlayerUnitTypeTwo + unitCountOtherPlayerUnitTypeTwo;
+
+  private Collection<Unit> getUnitsOfOtherPlayerOfUnitTypeOne() {

```

Comment:
Pls place helper methods below their first usage. For most of these it looks like that could be the case. Grouping them all at the top is not so fun, when reading the test methods I keep having to scroll back up to find the helper. 

Second, the private helper methods should be marked as static where possible.